terraform/infrastructure/modules/04-fargate/01-variables.tf

```diff
@@ -0,0 +1,44 @@
+variable "name" {
+  type = string
+}
+
+variable "domain_name" {
+  type = string
+}
+
+variable "tags" {
+  type = map(string)
+}
+
+variable "region" {
+  type = string
+}
+
+variable "private_dns_arn" {
+  type = string
+}
+
+variable "database_url" {
+  type      = string
+  sensitive = true
+}
+
+variable "private_subnet_ids" {
+  type = any
+}
+
+variable "private_subnet_objects" {
+  type = any
```

Comment:
try to avoid `any` types, especially without a description - it makes it impossible for a user to know what's expected. If you need a list of objects, you can make the full shape of the object explicit here.